### PR TITLE
ENH: InterestAreas object

### PR DESCRIPTION
**Pull request by teonbrooks**
This object reorganizes the data into a trial by interest area format to look at properties across trials with ease.
Test needed.

### CODE REVIEW

**Comment by Eric89GXL** [comment]
Please also add an example, since I have no idea what an interest area is :)

**Comment by Eric89GXL** [comment]
You'll need to get `OrderedDict` from mne-python

File: pyeparse/ia.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    """
+    def __init__(self, raw, ias, event_id=42, depmeas='fix'):
+        if isinstance(raw, string_types):
+            raw = read_raw(raw)
+        if not isinstance(raw, _BaseRaw):
+            raise TypeError('raw must be Raw instance of filename, not %s'
+                            % type(raw))
+
+        trial_ids = raw.find_events('TRIALID', 1)
+        self.n_trials = n_trials = trial_ids.shape[0]
```
Comment (dengemann) [inline pyeparse/ia.py]: should we stick with 'eopchs' for consistency with MNE?

File: pyeparse/ia.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+class IA(OrderedDict):
+    def __init__(self, entry):
+        super(IA, self).__init__(entry)
+
+    def __repr__(self):
+        return '<Interest Areas: %s>' % ', '.join(self.keys())
+
+
+def read_ia(filename):
+    ia = open(filename).readlines()
```
Comment (larsoner) [inline pyeparse/ia.py]: booo hissss
`with open`

**Comment by dengemann** [comment]
>Please also add an example, since I have no idea what an interest area is :)

+1 I have no idea what this is good for :) I'm curious! Sounds like an interesting area of development ... ;)

File: pyeparse/ia.py (hunk 1 above)
Comment (teonbrooks) [inline pyeparse/ia.py]: like `n_epochs`?

**Comment by teonlamont** [comment]
sure, sure :) examples are on the way. InterestAreas are at the center of lots of reading research and cognitive tasks in general. They are essentially the ROIs of reading. Lots of reading measures such as gaze duration, single fixation duration, first-fixation duration, first-pass reading, etc. are defined wrt a fixation or fixations, their order, and location. Gaze duration is defined as the sum of fixations from the first time you enter a region until you exit in any direction whereas go-past is defined as the sum of fixations from the first time you enter the region until you make a right-bound exit to the following word. 

There are lots of research that model these different measures to different cognitive proccesses but the backbone of them all are defined in Interest Areas.

For reading, they are usually rectangular bounding boxes around the words in the sentence. They are parsed with SR-Research Experiment Builder. I used their template as a basis.

**Comment by teonlamont** [comment]
real example with code to come.

example for my experiment I'm working on now:

trial

File: pyeparse/ia.py (hunk 2 above)
Comment (teonbrooks) [inline pyeparse/ia.py]: lol, what's the diff?

Comment (larsoner) [inline pyeparse/ia.py]: `with open():` will ensure the file gets closed when you are done with it. On Py3k you get `ResourceWarning`s in some cases if you don't do it.

Comment (teonbrooks) [inline pyeparse/ia.py]: ah, got it.

**Comment by teonlamont** [comment]
@Eric89GXL I tried `git grep OrderedDict` for mne-python I couldn't find any mentions

**Comment by dengemann** [comment]
from collections import OrderedDict

**Comment by teonlamont** [comment]
I already have that here: https://github.com/pyeparse/pyeparse/pull/49/files#diff-7489481660ad1bad504b865f7f0982bcR8 but the build crashed

**Comment by Eric89GXL** [comment]
we must not support 2.6 on mne-python, take it from `vispy` then, it uses a standard Python 2.6 replacement

**Comment by dengemann** [comment]
@teonlamont where are we here actually? Is this still alive?

**Comment by teonlamont** [comment]
yep, it is. it needs unit testing and sample data then it will be ready for review.

**Comment by teonbrooks** [comment]
closing PR. these scripts have been incorporated into the project-specific analysis scripts

**Pull request closed without merging**